support/ebpf/bpfdefs.h

```diff
@@ -61,89 +68,79 @@
-
-  // Sends `SIGTRAP` to the current task, killing it and capturing a coredump.
-  //
-  // Only use this in code paths that you expect to be hit by a very specific process that you
-  // intend to debug. Placing it into frequently taken code paths might otherwise take down
-  // important system processes like sshd or your window manager. For frequently taken cases,
-  // prefer using the `DEBUG_CAPTURE_COREDUMP_IF_TGID` macro.
-  //
-  // This macro requires linking against kernel headers >= 5.6.
-  #define DEBUG_CAPTURE_COREDUMP()                                                          \
-    ({                                                                                      \
-      /* We don't define `bpf_send_signal_thread` globally because it requires a      */    \
-      /* rather recent kernel (>= 5.6) and otherwise breaks builds of older versions. */    \
-      long (*bpf_send_signal_thread)(u32 sig) = (void *)BPF_FUNC_send_signal_thread;        \
-      bpf_send_signal_thread(SIGTRAP);                                                      \
+static void *(*bpf_map_lookup_elem)(void *map, void *key) = (void *)BPF_FUNC_map_lookup_elem;
+static int (*bpf_map_update_elem)(void *map, void *key, void *value, u64 flags) = (void *)
+    BPF_FUNC_map_update_elem;
+static int (*bpf_map_delete_elem)(void *map, void *key) = (void *)BPF_FUNC_map_delete_elem;
+static int (*bpf_probe_read)(void *dst, int size, const void *unsafe_ptr) = (void *)
+    BPF_FUNC_probe_read;
+static unsigned long long (*bpf_ktime_get_ns)(void) = (void *)BPF_FUNC_ktime_get_ns;
+static unsigned long long (*bpf_get_current_pid_tgid)(void) = (void *)BPF_FUNC_get_current_pid_tgid;
+static int (*bpf_get_current_comm)(void *buf, int buf_size) = (void *)BPF_FUNC_get_current_comm;
+static void (*bpf_tail_call)(void *ctx, void *map, int index) = (void *)BPF_FUNC_tail_call;
+static unsigned long long (*bpf_get_current_task)(void) = (void *)BPF_FUNC_get_current_task;
+static int (*bpf_perf_event_output)(
+    void *ctx, void *map, unsigned long long flags, void *data, int size) = (void *)
+    BPF_FUNC_perf_event_output;
+static int (*bpf_get_stackid)(void *ctx, void *map, u64 flags) = (void *)BPF_FUNC_get_stackid;
+static unsigned long long (*bpf_get_prandom_u32)(void) = (void *)BPF_FUNC_get_prandom_u32;
+
+__attribute__((format(printf, 1, 3))) static int (*bpf_trace_printk)(
+    const char *fmt, int fmt_size, ...) = (void *)BPF_FUNC_trace_printk;
+
+static long (*bpf_probe_read_user)(void *dst, int size, const void *unsafe_ptr) = (void *)
+    BPF_FUNC_probe_read_user;
+static long (*bpf_probe_read_kernel)(void *dst, int size, const void *unsafe_ptr) = (void *)
+    BPF_FUNC_probe_read_kernel;
+
+  // The sizeof in bpf_trace_printk() must include \0, else no output
+  // is generated. The \n is not needed on 5.8+ kernels, but definitely on
+  // 5.4 kernels.
+  #define printt(fmt, ...)                                                                         \
```

Comment:
In general, I like to have indentation for nested CPP directives. But it is not consistent with the (none-) indentation for the function definitions above.

Reply:
Didn't find a configuration for this, resolving.

Reply:
Whatever the decision is, not a blocker for me. Just in general, I like indentation for nested code blocks =) - and I see CPP directives as (meta-)code.
More important is to avoid the tabs vs spaces discussion 😆